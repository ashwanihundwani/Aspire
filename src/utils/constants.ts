import { windowWidth, windowHeight } from "./utils";
import { Platform } from "react-native";

// Tab Name Constants
export const Tabs = { 
    home: "Home", 
    debit: "Debit Card", 
    payments: "Payments",
    credit: "Credit",
    account: "Profile"
} as const;

// Stack Route/Name Constants
export const StackScreens = { 
    debitCardTab: "Tab", 
    weeklySpentLimit: "WeeklySpendingLimit", 
    
} as const;

// Font Name Constants
export const FontNames = { 
    Avenir: "AvenirNext"  
} as const;

export const AppColors = {
    lightGray: "#DDDDDD",
    lightGreen: "#01D167",
    navigationBackColor: "#0C365A",
    navigationTintColor: "#FFFFFF",
    whiteColor: "#FFFFFF",
    darkTextColor: "#22222266",
    descriptionColor: "#222222",
    spentLimitText: "#22222233",
    separatorColor: "#DDDDDD",
    lightGreenWithOpacity: 'rgba(1,209,103, 0.07)',
    debitCardTitleColor: "#25345F",
    disabledSaveButton: "#EEEEEE"
}

export const CommonString = {
    currencySign: "S$",
    dollarSign: "$",
    switchSuffix:"_Switch"
}

export const DebitCardString = {
    headerTitle: "Debit Card",
    expiry: "Thru: ",
    cvv: "CVV: ",
    showCard: "Show card number",
    hideCard: "Hide card number",
    availableBalance: "Available Balance",
    spendLimitLabel: "Debit card spending limit",
    cardInfo: {
        cardNumberSplitter:"-",
        hiddenCVV: "* * *",
        dotString: '\u2022\u2022\u2022\u2022'
    }
}

export const WeeklySpendingString = {
    headerTitle: "Spending Limit",
    descriptionMessage: "Set a weekly debit card spending limit",
    infoMessage: "Here weekly means the last 7 days - not the calendar week", 
    saveButton: "Save",
    buttonTitle1: "S$ 5,000",
    buttonTitle2: "S$ 10,000",
    buttonTitle3: "S$ 20,000",
}

export const DebitCardOptions = {
    Titles : {
        topup: "Top-up account",
        weeklyspent: "Weekly spending limit",
        freezeCard: "Freeze card",
        newCard: "Get a new card",
        deactivated: "Deactivated cards"
    },
    Descriptions: {
        topup: "Deposit money to your account to use with card",
        weeklyspent: "You havenâ€™t set any spending limit on card",
        freezeCard: "Your debit card is currently active",
        newCard: "This deactivates your current debit card",
        deactivated: "Your previously deactivated cards"
    },
}

export const layouting = {
    // Standard margin calculation to accomodate various screen sizes accross devices.
    spacingFactor: 0.058 * windowWidth(),
}

export const testIDs = {
    showHideButton: "DebitCard.CardInfo.showHide",
    cardOption: "DebitCard.CardOption",
    spendProgress:"DebitCard.SpendProgress"
}

  