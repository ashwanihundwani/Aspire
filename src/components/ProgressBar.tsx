import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { windowWidth } from '../utils/utils'

interface ProgressBarProps {
    tintColor: string,
    backgroundColor: string,
    width: number,
    height: number,
    percentage: number
    borderRadius: number
}

const ProgressBar: React.FC<ProgressBarProps> = (props: ProgressBarProps) => {

    const containerStyle = {
        width: props.width,
        height: props.height,
        borderRadius: 15,
        backgroundColor: props.backgroundColor
    }
    const trapezoidStyle = {

        width: props.width * props.percentage / 100,
        height: 20,
        borderTopColor: props.tintColor,
        borderTopWidth: 40,
        borderRightWidth: 20,
        borderRightColor: 'transparent',
    }
    const solidStyle = {
        width: "100%",
        backgroundColor: props.tintColor,
        height: 20
    }

    return <View style={[containerStyle, { overflow: "hidden" }]}>
        <View style={props.percentage < 100 ? trapezoidStyle : solidStyle}>
        </View>

    </View>
}

export default React.memo<ProgressBarProps>(ProgressBar)

const styles = StyleSheet.create({
})