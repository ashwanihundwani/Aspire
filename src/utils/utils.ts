
import { FontNames } from './constants'
import { Dimensions } from 'react-native'

export enum FontStyle {
    Regular = "Regular",
    Medium = "Medium",
    DemiBold = "DemiBold",
    Bold = "Bold",
}

export function avenirFont(style: FontStyle) {
    return FontNames.Avenir + "-" + style
}

export function windowWidth() {
    return Dimensions.get("window").width
}

export function windowHeight() {
    return Dimensions.get("window").height
}

export function parseNumberFromString(str: string) {
    return parseInt(str.replace(/,/g, ''), 10);
}