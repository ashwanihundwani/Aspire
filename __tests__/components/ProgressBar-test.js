/**
 * @format
 */

import 'react-native';
import React from 'react';
import ProgressBar from '../../src/components/ProgressBar';
import renderer from 'react-test-renderer';
import { AppColors, layouting } from '../../src/utils/constants'
import {windowWidth } from '../../src/utils/utils'


it('renders correctly', () => {
    const tree = renderer
      .create( <ProgressBar
        backgroundColor={AppColors.lightGreenWithOpacity}
        tintColor={AppColors.lightGreen}
        width={windowWidth() - (layouting.spacingFactor * 2)}
        height={15}
        percentage={80}
        borderRadius={8}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});