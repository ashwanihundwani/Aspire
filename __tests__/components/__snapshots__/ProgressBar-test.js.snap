// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgba(1,209,103, 0.07)",
        "borderRadius": 15,
        "height": 15,
        "width": 663,
      },
      Object {
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      Object {
        "borderRightColor": "transparent",
        "borderRightWidth": 20,
        "borderTopColor": "#01D167",
        "borderTopWidth": 40,
        "height": 20,
        "width": 530.4,
      }
    }
  />
</View>
`;
